Part1/code/Dog001/Dog001.ino

```diff
@@ -1,141 +1,147 @@
-            kneeAngle2a = kneeAngle - KNEEACTANGLE;                       // take away known angle
-            kneeAngle2 = (kneeAngle2a * 71) / 4068;                        // convert degrees to radians
-
-            kneeAngle3 = asin ( (KNEEROD2 * sin(kneeAngle2)/KNEEROD));
-            kneeAngle3a = (kneeAngle3 * 4068 / 71);                        // convert radians to degrees
-
-            kneeAngle4a = 180 - kneeAngle2a - kneeAngle3a;                  // we now know all four angles
-            kneeAngle4 = (kneeAngle4a * 71) / 4068;                        // convert degrees to radians
-
-            kneeActuator = (((sin(kneeAngle4)) * KNEEROD)) / sin(kneeAngle2);
-
-            Serial.print(legLength);
-            Serial.print(" , ");
-            Serial.println(kneeActuator);      
-          
-
-       }  // end of timed event
-
-}
+#include <EasyTransfer.h>
+#include <Math.h>
+
+//create object
+EasyTransfer ET1;   // send serial
+EasyTransfer ET2;   // rec serial
+
+// knee calcs
+#define DIGITLENGTH 330L    // length of each top/bottom leg
+#define KNEEROD 180L       // length of push rod
+#define KNEEROD2 94L        // other side of push rod triangle
+#define KNEEACTANGLE 15L   // angle between actuator and upp leg joint
+
+double legLength;           // required overall leg length
+double kneeAngle;           // the actual angle of the knee between top and bottom sections
+double kneeAngle2;          // angle between bottom of leg and actuator
+double kneeAngle2a;          // angle between bottom of leg and actuator
+double kneeAngle3;          // other angle
+double kneeAngle3a;          // other angle
+double kneeAngle4;          // other angle.
+double kneeAngle4a;          // other angle
+double kneeActuator;        // calculated length of actuator from joint
+
+int axis1;
+int axis2;
+int axis3;
+int axis4;
+int axis5;
+int axis6;
+
+int mode;
+int count;
+
+int menuFlag = 0;        
+
+struct RECEIVE_DATA_STRUCTURE{
+  //put your variable definitions here for the data you want to send
+  //THIS MUST BE EXACTLY THE SAME ON THE OTHER ARDUINO
+  
+    bool menuDown;      
+    bool Select; 
+    bool menuUp;  
+    bool toggleBottom;  
+    bool toggleTop; 
+    int mode;  
+    int16_t RLR;
+    int16_t RFB;
+    int16_t RT;
+    int16_t LLR;
+    int16_t LFB;
+    int16_t LT;
+};
+
+struct SEND_DATA_STRUCTURE{
+  //put your variable definitions here for the data you want to receive
+  //THIS MUST BE EXACTLY THE SAME ON THE OTHER ARDUINO
+  int16_t mode;
+  int16_t count;
+};
+
+SEND_DATA_STRUCTURE mydata_send;
+RECEIVE_DATA_STRUCTURE mydata_remote;
+
+unsigned long previousMillis = 0;
+const long interval = 20;
+
+long previousSafetyMillis;
+
+void setup() {
+
+  Serial.begin(57600);
+  Serial2.begin(57600);
+
+  ET1.begin(details(mydata_send), &Serial2);
+  ET2.begin(details(mydata_remote), &Serial2);
+
+
+}
+
+
+float radToDeg(radians) {
+  return radians * 4068/71

```

Comment:
This line doesn't compile either since it is missing the semicolon at the end of the line. The body of degToRad() is missing a semicolon too.